ddtrace/profiling/collector/_traceback.pyx

```diff
@@ -51,9 +92,18 @@ cpdef pyframe_to_frames(frame, max_nframes):
     nframes = 0
     while frame is not None:
         nframes += 1
-        if len(frames) < max_nframes:
-            code = frame.f_code
-            lineno = 0 if frame.f_lineno is None else frame.f_lineno
-            frames.append((code.co_filename, lineno, code.co_name, _extract_class_name(frame)))
-        frame = frame.f_back
+        # Python 3.11 moved PyFrameObject to internal C API and cannot be directly accessed from tstate
+        IF PY_MAJOR_VERSION > 3 or (PY_MAJOR_VERSION == 3 and PY_MINOR_VERSION >= 11):
+            if len(frames) < max_nframes:
+                code = PyFrame_GetCode(<PyFrameObject*> frame)
```

Comment:
```suggestion
                code = <object>PyFrame_GetCode(<PyFrameObject*> frame)
```

or maybe `PyCodeObject*`, and then define the properties on the cdef extern above so it knows about `co_name`/`co_filename` ?